lib/wlib/strings/DynamicString.h

```diff
@@ -0,0 +1,476 @@
+	 */
+	DynamicString(const char* str){
+		uint16_t size = (uint16_t) strlen(str);
+		m_buffer = (char *)memory_alloc((size + 1) * sizeof(char));
+		strncpy(m_buffer, str, size);
+		m_buffer[size] = '\0';
+	}
+
+	/**
+	 * Constructor creates string using DynamicString object
+	 *
+	 * @param str @code DynamicString object
+	 */
+	DynamicString(const DynamicString &str){
+		DynamicString(str.c_str());
+	}
+
+	/**
+	 * Assign operator assigns current object to given object
+	 *
+	 * @param str @code DynamicString object
+	 * @return current object
+	 */
+	DynamicString operator= (const DynamicString &str){
+		return operator=(str.c_str());
+	}
+
+	/**
+	 * Assign operator assigns current object to given character string
+	 * @param str
+	 * @return current object
+	 */
+	DynamicString operator= (const char* str){
+		uint16_t size = (uint16_t) strlen(str);
+		m_buffer = (char *)memory_realloc(m_buffer, (size + 1) * sizeof(char));
+		strncpy(m_buffer, str, size);
+		m_buffer[size] = '\0';
+
+		return *this;
+	}
+
+	/**
+	 * Provides current length of string
+	 *
+	 * @return string length
+	 */
+	uint16_t length() const{
+		uint16_t i = 0;
+		while (m_buffer[i] != 0) i++;
+		return i;
+	}
+
+	/**
+	 * Clears the string such that there are no characters left in it
+	 */
+	void clear(){
+		m_buffer[0] = '\0';
+	}
+
+	/**
+	 * Element access operator gives access to character at @code pos
+	 *
+	 * @param pos the position of the character
+	 * @return character at @code position
+	 */
+	char &operator[](uint16_t pos){
+		return m_buffer[pos];
+	}
+
+	/**
+	 * Element access operator gives access to character at @code pos.
+	 * Character is constant
+	 *
+	 * @param pos the position of the character
+	 * @return character at @code position
+	 */
+	const char &operator[](uint16_t pos) const{
+		return m_buffer[pos];
+	}
+
+	/**
+	 * Provides access to character at @code pos
+	 *
+	 * @param pos the position of the character
+	 * @return character at @code position
+	 */
+	char &at(uint16_t pos){
+		return m_buffer[pos];
+	}
+
+	/**
+	 * Provides access to character at @code pos. Character is constant
+	 *
+	 * @param pos the position of the character
+	 * @return character at @code position
+	 */
+	const char &at(uint16_t pos) const{
+		return m_buffer[pos];
+	}
+
+	/**
+	 * Checks if string is empty or not
+	 *
+	 * @return if string is empty or not
+	 */
+	bool empty() const {
+		return length() == 0;
+	}
+
+	/**
+	 * Provides access to the first character in the string
+	 *
+	 * @return the first character
+	 */
+	char &front(){
+		return m_buffer[0];
+	}
+
+	/**
+	 * Provides access to the first character in the string. Character is constant
+	 *
+	 * @return the first character
+	 */
+	const char &front() const{
+		return m_buffer[0];
+	}
+
+	/**
+	 * Provides access to the last character in the string
+	 *
+	 * @return the last character
+	 */
+	char &end(){
+		if (empty()) return m_buffer[0];
+
+		return m_buffer[length() - 1];

```

Comment:
Accessing the last element is current an `O(n)` operation! (Same with `empty`)

Reply:
Ya I'll fix this with the length variable